libgraphics/src/path.cpp

```diff
@@ -131,12 +133,19 @@ void MCGPathCopy(MCGPathRef self, MCGPathRef& r_new_path)
 			t_new_path -> is_mutable = false;
 			r_new_path = t_new_path;
 		}
+		else
+		{
+			MCGPathDestroy(t_new_path);
```

Comment:
Would it be easier to manage cleanup in the MCGPath* functions if you introduced an `MCGAutoPathRef` or some equivalent of `MCAutoValueRefBase<>`?